### DISCUSSION TITLE
Performance improvement idea - shorten default History view

### DISCUSSION THREAD

**Discussion by randyoo** [home-assistant/frontend#17058] [Other feature requests]
Recently, some performance improvements were made to the History view that's shown when an entity is tapped/clicked. It's populated with a chart or view showing the past 24 hours, and performance is snappy. If the "Show More" option is chosen, though, the history page is opened, defaulting to that entity's history for the previous day at midnight through the end of the present day.

On lower performance hardware, and especially with entities that have a high sample/recorder rate, **this can cause a delay of many seconds as the history chart is built**. The date selector remains disabled/locked out during this period, preventing the user from selecting the historical range of interest, which might only be a shorter period of minutes that could have loaded quickly.

Suggest the following to improve user experience:

1. Remove the UI lockout for the range selection on initial load after clicking "Show more" -or-
2. Default to a shorter time period for the initial display after clicking "Show more": perhaps the past 3 hours? Or, mirroring the rolling 24 hours shown in the entity view might yield better performance, if there's caching taking place.
3. Add a few more options in the left column of the range selection. It's presently "Today, Yesterday, This Week, Last Week". Suggest adding "Past hour, 3 hours, 6 hours, 12 hours", similar to what Victron does with VRM. (Demo can be viewed [here](https://vrm.victronenergy.com/))

**Comment by randyoo** [comment]
In regards to point 3 (above), here is a screenshot of Victron's VRM options:
![Screenshot 2023-06-27 at 11 31 31 PM](https://github.com/home-assistant/frontend/assets/7457804/0cd9d46b-2a31-489d-bf48-649ec5aecb25)

Obviously, with recorder retention defaulting to 2 weeks, many of those options wouldn't apply.

**Comment by Xitee1** [comment]
If this gets added we need a (user specific) setting for the default option. I use the histoy really often to see how the value was 24h ago (at the same time yesterday) (for example temperatures). Needing extra click(s) for this each time to set it from 3h to 24h would really bother me.

**Reply by randyoo** [reply]
Thanks for the feedback, I find myself doing the same very often. The default rolling 24 hour window on initial history graph (shown when an entity is selected) is very useful, and like you, I use it very frequently to compare present values to 24 hours earlier. Agreed it would be frustrating to change this behavior.

But, to be clear, my suggestion was related to lengthy delays encountered **AFTER** choosing "Show more" on the entity details view, owing to the default of showing the previous day, starting at midnight, up until the present time. If that view also defaulted to a rolling 24 hours, it might be more performant, since the same chart was already built and displayed on the initial history window.

I'll try to revise my suggestion to make it less ambiguous.

**Reply by Xitee1** [reply]
Ah I thought you meant the graph before clicking on "Show more".
That would be a good idea. I also have slow loading times with that. And most of the time I only click on "Show more" to get a non-smoothen out graph without mean values and change it to 24h after it has been loaded a range of around 2 days (Actually most of the time I use the "Show more" option to show less 😂).
Before the the default graphs used mean values it made more sense to show a extended time range. But now that it has been changed some updates ago it should be better the same time range and just with more details.
